math/matrix/samedimensions.go

```diff
@@ -0,0 +1,18 @@
+package matrix
+
+import "fmt"
+
+// SameDimensions checks if two matrices have the same dimensions.
+func (m *Matrix[T]) SameDimensions(matrix2 Matrix[T]) (bool, error) {
```

Comment:
I'm kind of on the fence about the naming of this method. Do you think that `MatchDimensions` or `VerifySameDimensions` are better names when reading the code through?